### PR TITLE
[DNM]fix region cache

### PR DESCRIPTION
**Pull request by morefreeze**
PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
@disksing @siddontang PTAL

File: kv.go [hunk 1]
```diff
@@ -177,6 +177,10 @@ func (s *tikvStore) SendKVReq(req *pb.Request, region *requestRegion) (*pb.Respo
 		resp *pb.Response
 	)
 	for tryCount := 1; tryCount <= maxSendRetry; tryCount++ {
+		region, err = s.getRegion(region.GetLookupKey())
```
Comment (qiuyesuifeng) [inline kv.go]: The input variable region seems useless?

File: kv.go [hunk 2]
```diff
@@ -207,33 +211,21 @@ func (s *tikvStore) SendKVReq(req *pb.Request, region *requestRegion) (*pb.Respo
...
+			s.regionCache.UpdateLeader(leaderInfo.GetRegionId(), leaderInfo.GetLeader().GetId())
 			continue
 		}
 		if tryCount > 1 {
 			log.Debugf("Get response in retrying %d times", tryCount)
 		}
 		return resp, nil
 	}
-	return resp, errors.Trace(err)
+	return nil, errors.New("max retry")
```
Comment (siddontang) [inline kv.go]: use a better error information.

File: kv.go (hunk 1 above)
Comment (morefreeze) [inline kv.go]: Put here is for easy implement. Refactor retry later.

File: kv.go (hunk 2 above)
Comment (morefreeze) [inline kv.go]: `"Send has reached maximum retry limit[%d].", maxSendRetry`?

**Comment by morefreeze** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**